src/ui/server/helpers/construct-payload/index.ts

```diff
@@ -0,0 +1,36 @@
+import { RequestBody } from '../../../types';
+import { objectHasKeysAndValues } from '../object';
+import { isPopulatedArray } from '../array';
+
+/**
+ * constructPayload
+ * contructs payload from requestBody
+ * filters out anything which is not part of specified fieldIds
+ * @param {RequestBody} requestBody
+ * @param {Array<String>} fieldIds
+ * @returns {Object} constructed payload
+ */
+const constructPayload = (requestBody: RequestBody, fieldIds: Array<string>) => {
+  let payload = {};
+
+  // if either requestBody or fieldsIds are empty, return empty object
+  if (!objectHasKeysAndValues(requestBody) || !isPopulatedArray(fieldIds)) {
+    return payload;
+  }
```

Comment:
It looks like this condition might not be required  because they are required params. At least for the `fieldIds`? I maybe wrong though.

Reply:
I think you can pass an empty array too

Reply:
Am I missing something, why would a payload from a form be an array?

Reply:
fixed

Reply:
Nice one - comment is invalid now though it mentions `requestBody` but it only cares about the array